Pointwise.lua

```diff
@@ -31,11 +31,14 @@ function Pointwise:createIODescriptors(input)
    end
 
    local nElem = input:nElement()
-   self.nElem = self.nElem or nElem -- this goes to the second branch only once
-   if self.iDesc and nElem == self.nElem then return end
-   self.nElem = nElem
-   self.iDesc = cudnn.toDescriptor(input:view(1,1,1,nElem))
-
+   self.nElem = self.nElem or nElem
+   if self.iDesc and nElem == self.nElem and
```

Comment:
If the dimensions of the input tensor are changed, or (new) the type is changed, a new descriptor is created. If both these properties are the same, the same descriptor can be reused. Except for the type part it should be exactly the same as before.